manifests/repomd_signatures.pp

```diff
@@ -0,0 +1,12 @@
+# Generate or regenerate Yum repository metadata GPG signatures
+class pulp::repomd_signatures {
+  if $::pulp::yum_regenerate_repomd_signatures {
+    exec { 'yum_regenerate_repomd_signatures':
+      command => "/usr/bin/sudo -u apache -H /usr/bin/bash -c 'cd /var/lib/pulp/published/yum/master/yum_distributor/ 2>/dev/null || exit 0 ; for d1 in `ls` ; do pushd \$d1 ; for d2 in `ls` ; do pushd \$d2/repomd >/dev/null ; gpg --yes --detach-sign --armor repomd.xml ; popd >/dev/null ; done ; popd >/dev/null ; done'",
```

Comment:
Please check which attributes would be applicable here https://puppet.com/docs/puppet/5.5/types/exec.html#exec-attributes
At first glance, `user`, `cmd` would probably be helpful.

Also putting it in a multiline string would make it a lot easier to read.

Reply:
Hmmm...

I can probably use `cwd` here, and I can certainly make it a multiline string.

However, I can't replace the sudo call with a `user` attribute because `gpg` needs `$HOME` to be set properly, but puppet doesn't set `$HOME` (https://projects.puppetlabs.com/issues/5224)...  So I need to use `sudo -H` to ensure that `$HOME` is set properly.

Reply:
You can set HOME via the env parameter of exec.

Reply:
It is true that you can hard-code a value for HOME via the env parameter of exec.  However, if the user sets a different home directory for the apache user in /etc/passwd (or if the user tries to install foreman on a different Linux distribution that uses a different home directory for apache), then this hard-coded value will be wrong, while `sudo -H` will pick up the correct value from /etc/passwd.